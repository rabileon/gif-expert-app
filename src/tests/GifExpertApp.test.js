import React from 'react';
const { shallow } = require('enzyme');
const { GifExpertApp } = require('../GifExpertApp');

describe('Prueba en <GifExpertApp/>', () => {
  test('debe mostrarse correctament', () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de mostrar una lista de categorias', () => {
    const categories = ['TREDING'];

    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
});
