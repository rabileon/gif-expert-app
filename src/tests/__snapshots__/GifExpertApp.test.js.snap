// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifExpertApp/> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h1
    className="title-app"
  >
    Gif Expert App
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="TRENDING"
      key="TRENDING"
    />
  </ol>
</Fragment>
`;

exports[`Prueba en <GifExpertApp/> debe mostrarse correctament 1`] = `
<Fragment>
  <h1
    className="title-app"
  >
    Gif Expert App
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="TRENDING"
      key="TRENDING"
    />
  </ol>
</Fragment>
`;
